scipy/interpolate/_bary_rational.py

```diff
@@ -560,5 +511,203 @@ def clean_up(self, cleanup_tol=1e-13):
         self._poles = None
         self._residues = None
         self._roots = None
-        
+
         return ni
+
+    def poles(self):
+        """Compute the poles of the rational approximation.
+
+        Returns
+        -------
+        poles : array
+            Poles of the AAA approximation, repeated according to their multiplicity
+            but not in any specific order.
+        """
+        if self._poles is None:
+            # Compute poles via generalized eigenvalue problem
+            m = self.weights.size
+            B = np.eye(m + 1, dtype=self.weights.dtype)
+            B[0, 0] = 0
+
+            E = np.zeros_like(B, dtype=np.result_type(self.weights,
+                                                      self._support_points))
+            E[0, 1:] = self.weights
+            E[1:, 0] = 1
+            np.fill_diagonal(E[1:, 1:], self._support_points)
+
+            pol = scipy.linalg.eigvals(E, B)
+            self._poles = pol[np.isfinite(pol)]
+        return self._poles
+
+    def residues(self):
+        """Compute the residues of the poles of the approximation.
+
+        Returns
+        -------
+        residues : array
+            Residues associated with the `poles` of the approximation
+        """
+        if self._residues is None:
+            # Compute residues via formula for res of quotient of analytic functions
+            with np.errstate(divide="ignore", invalid="ignore"):
+                N = (1/(np.subtract.outer(self.poles(), self._support_points))) @ (
+                    self._support_values * self.weights
+                )
+                Ddiff = (
+                    -((1/np.subtract.outer(self.poles(), self._support_points))**2)
+                    @ self.weights
+                )
+                self._residues = N / Ddiff
+        return self._residues
+
+    def roots(self):
+        """Compute the zeros of the rational approximation.
+
+        Returns
+        -------
+        zeros : array
+            Zeros of the AAA approximation, repeated according to their multiplicity
+            but not in any specific order.
+        """
+        if self._roots is None:
+            # Compute zeros via generalized eigenvalue problem
+            m = self.weights.size
+            B = np.eye(m + 1, dtype=self.weights.dtype)
+            B[0, 0] = 0
+            E = np.zeros_like(B, dtype=np.result_type(self.weights,
+                                                      self._support_values,
+                                                      self._support_points))
+            E[0, 1:] = self.weights * self._support_values
+            E[1:, 0] = 1
+            np.fill_diagonal(E[1:, 1:], self._support_points)
+
+            zer = scipy.linalg.eigvals(E, B)
+            self._roots = zer[np.isfinite(zer)]
+        return self._roots
```

Comment:
I decided to move these to AAA only as for FH they aren't that useful:
```
julia> using BaryRational

julia> x = range(0, 1, 51)
0.0:0.02:1.0

julia> fh = FHInterp(x, sin.(x), order=3)
(::FHInterp{Vector{Float64}}) (generic function with 1 method)

julia> prz(fh.x, fh.f, fh.w)[3]
50-element Vector{ComplexF64}:
                 0.0 + 0.0im
   8.476977910743983 + 0.0im
  -2.150503719495504 + 0.0im
  2.8540256782515683 + 0.0im
  0.4992401435746384 + 0.4806751013396058im
  0.4992401435746385 - 0.48067510133960584im
  0.0698648894186103 + 0.05024696155519378im
 0.06986488941861031 - 0.05024696155519378im
  0.9301351271397813 + 0.050246890044132894im
  0.9301351271397813 - 0.050246890044132894im
 0.11524735870316144 + 0.06222351512756596im
 0.11524735870316144 - 0.06222351512756596im
  0.8847528926684755 + 0.06222336714078676im
  0.8847528926684756 - 0.06222336714078676im
 0.15802957767784784 + 0.07005015687811553im
 0.15802957767784784 - 0.07005015687811554im
  0.8419711752500212 + 0.07005005621554712im
  0.8419711752500213 - 0.07005005621554712im
 0.19979893057138223 + 0.07594789128033365im
 0.19979893057138226 - 0.07594789128033365im
  0.8002026565275886 + 0.07594806766726686im
  0.8002026565275887 - 0.07594806766726686im
  0.2410246302981232 + 0.08069981669334382im
                     ⋮
  0.7181129880456163 + 0.08468490610525051im
  0.7181129880456163 - 0.08468490610525051im
 0.32246535061567777 + 0.08810745842824674im
  0.3224653506156779 - 0.08810745842824674im
  0.6775417610373833 + 0.08811085104607308im
  0.6775417610373833 - 0.0881108510460731im
 0.36273346680911955 + 0.09108978623876343im
 0.36273346680911955 - 0.09108978623876342im
  0.6372771224424681 + 0.09109507481297181im
  0.6372771224424681 - 0.09109507481297181im
 0.40261254277632225 + 0.09366054451328429im
 0.40261254277632225 - 0.0936605445132843im
  0.5974028421584526 + 0.09366747076463088im
  0.5974028421584527 - 0.09366747076463088im
 0.44196965721926085 + 0.09571181615544264im
 0.44196965721926085 - 0.09571181615544264im
  0.5580516756058949 + 0.09571859118443778im
   0.558051675605895 - 0.09571859118443776im
   0.519268847503646 + 0.09693003994795131im
   0.519268847503646 - 0.09693003994795131im
  0.4807573875358231 + 0.09692702543737695im
  0.4807573875358231 - 0.09692702543737697im
```
ie. you get as many zeros as points

Reply:
Could readd in the future though

Reply:
Like extrapolation, filtering these is not straight forward as complex inputs are valid

Reply:
I don't understand I've to admit. Having `zeros`, `poles` and `residuals` feels like a feature common to all barycentric interpolators, so havign them for some and not others is strange.

Reply:
I changed my mind on this and reverted this